confgenerator/confgenerator.go

```diff
@@ -209,35 +211,49 @@ func generateOtelServices(receiverNameMap map[string]string, exporterNameMap map
 		if strings.HasPrefix(pID, "lib:") {
 			return nil, fmt.Errorf(`pipeline id prefix 'lib:' is reserved for pre-defined pipelines. Pipeline ID %q is not allowed.`, pID)
 		}
+
 		p := pipelines[pID]
+		// if the pipeline is default_pipeline, then split the pipeline into different streams.
+		if pID == "default_pipeline" {
+			for _, rID := range p.Receivers {
+				var pipelineID string
+				var defaultProcessors []string
+				if rID == "hostmetrics" {
```

Comment:
The receiver ids are arbitrary strings, right? Should we check against the receiver type instead of just the ID?

Reply:
Good catch